File: backend/app/schemas/analysis.py
from pydantic import BaseModel


class PriceStats(BaseModel):
    median: float
    avg: float
    min: float
    max: float
    count: int


class PriceSuggestion(BaseModel):
    suggested_price: float
    discount_pct: float
    price_position: str  # "above_market" | "at_market" | "below_market"


class StockSignal(BaseModel):
    level: str  # "in_stock" | "low_stock" | "out_of_stock" | "unknown"
    severity: str  # "ok" | "warning" | "critical" | "info"
    action: str | None = None


class AnalysisResponse(BaseModel):
    asin: str
    current_price: float | None
    price_stats: PriceStats | None
    suggestion: PriceSuggestion | None
    stock_signal: StockSignal


class InsightsResponse(BaseModel):
    asin: str
    price_analysis: str
    discount_recommendation: str
    competitive_positioning: str

File: frontend/services/analysis.ts
import { requestJson } from "./http";

export type PriceStats = {
  median: number;
  avg: number;
  min: number;
  max: number;
  count: number;
};

export type PriceSuggestion = {
  suggested_price: number;
  discount_pct: number;
  price_position: "above_market" | "at_market" | "below_market";
};

export type StockSignal = {
  level: "in_stock" | "low_stock" | "out_of_stock" | "unknown";
  severity: "ok" | "warning" | "critical" | "info";
  action: string | null;
};

export type AnalysisResult = {
  asin: string;
  current_price: number | null;
  price_stats: PriceStats | null;
  suggestion: PriceSuggestion | null;
  stock_signal: StockSignal;
};

export type InsightsResult = {
  asin: string;
  price_analysis: string;
  discount_recommendation: string;
  competitive_positioning: string;
};

export async function getAnalysis(asin: string): Promise<AnalysisResult> {
  return requestJson<AnalysisResult>(`/api/analysis/${encodeURIComponent(asin)}`);
}

export async function getAiInsights(asin: string): Promise<InsightsResult> {
  return requestJson<InsightsResult>(
    `/api/analysis/${encodeURIComponent(asin)}/insights`,
    { method: "POST" },
  );
}
